src/cargo/util/config/mod.rs

```diff
@@ -2106,16 +2107,22 @@ impl ConfigValue {
 
     /// Merge the given value into self.
     ///
-    /// If `force` is true, primitive (non-container) types will override existing values.
-    /// If false, the original will be kept and the new value ignored.
+    /// If `force` is true, primitive (non-container) types will override existing values
+    /// of equal priority. For arrays, incoming values of equal priority will be placed later.
```

Comment:
The code looks correct. Just what to make sure we have test to verify this behavior.